File: 0704-binary-search/0704-binary-search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int hi = nums.size() - 1, lo = 0, mid;
            
        while(lo <= hi)
        {
            mid = lo + (hi - lo) / 2;
            
            if(nums[mid] == target)
                return mid;
            if(nums[mid] > target)
                hi = mid - 1;
            else lo = mid + 1;
        }
        return -1;
    }
};

File: 0704-binary-search/0704-binary-search.java
class Solution {
    public int search(int[] arr, int target) {
        
        int n = arr.length;
        
        int start = 0;
        int end = n - 1;
        int ans = -1;
        
        while(start <= end)
        {
            int mid = (start + end) / 2;
            
            if(arr[mid] == target)
                return mid;
            
            if(arr[mid] > target)
            {
                end = mid - 1;
            }
            else if(arr[mid] < target)
            {
                start = mid + 1;
            }
            
        }
        
        
        return -1;
    }
}